File: ContosoWebFormsScaffolding/Samples.CS/4_Enumerations/Employee.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;

namespace Samples.Enumerations
{
    public class Employee
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }

        [Required]
        [DataType(DataType.EmailAddress)]
        public string Email { get; set; }

        [DataType(DataType.Url)]
        [Display(Name="Home Page URL:")]
        public string HomePage { get; set; }

        [Required(ErrorMessage="You must enter a salary level")]
        public SalaryLevel Salary { get; set; }

        // selecting a region is optional
        public Region? Region { get; set; }
    }
}

File: ContosoWebFormsScaffolding/Samples.VB/4_Enumerations/Employee.vb
Imports System.ComponentModel.DataAnnotations.Schema
Imports System.ComponentModel.DataAnnotations

Public Class Employee
    <DatabaseGenerated(DatabaseGeneratedOption.Identity)>
    Public Property Id As Guid

    <Required>
    <DataType(DataType.EmailAddress)>
    Public Property Email As String

    <DataType(DataType.Url)>
    <Display(Name:="Home Page URL")>
    Public Property HomePage As String

    <Required(ErrorMessage:="You must enter a salary level")>
    Public Property Salary As SalaryLevel

    ' selecting a region is optional
    Public Property Region As Region?

End Class
